### PR TITLE
TPS-228 Created DayOffType endpoints.

### PR DESCRIPTION
**Pull request by biruk-telonai**
Created DayOffType endpoints.

### CODE REVIEW

File: Telonai/TelonaiBFF/Controllers/DayOffRequestController.cs [hunk 1]
```diff
@@ -16,13 +16,15 @@ public class DayOffRequestController : ControllerBase
     private readonly IDayOffRequestService<DayOffRequestModel, DayOffRequest> _service;
     private readonly ILogger<DayOffRequestController> _logger;
     private readonly IScopedAuthorization _scopedAuthrorization;
+    private readonly IDayOffService _dayOffTypeService; 
 
     public DayOffRequestController(IDayOffRequestService<DayOffRequestModel, DayOffRequest> service, 
-        ILogger<DayOffRequestController> logger, IScopedAuthorization scopedAuthrorization)
+        ILogger<DayOffRequestController> logger, IScopedAuthorization scopedAuthrorization, IDayOffService dayOffService)
```
Comment (samsonkassa1) [inline Telonai/TelonaiBFF/Controllers/DayOffRequestController.cs:22]: This should named dayOffTypeService

File: Telonai/TelonaiBFF/Entities/IdValuePair.cs [hunk 2]
```diff
@@ -26,5 +26,4 @@ public class CheckedBoxSixteenType : IdValuePair { }
 public class QuarterType : IdValuePair { }
 public class ExemptFromFutaTaxType : IdValuePair { }
 public class BankAccountType : IdValuePair { }
-public class DayOffType : IdValuePair { }
```
Comment (samsonkassa1) [inline Telonai/TelonaiBFF/Entities/IdValuePair.cs:29]: Why do you delete this?

File: Telonai/TelonaiBFF/Helpers/AutoMapperProfile.cs [hunk 3]
```diff
@@ -381,11 +381,11 @@ public AutoMapperProfile()
 
         CreateMap<DayOffRequestModel, DayOffRequest>()
             .ForMember(dest => dest.Id, opt => opt.Ignore())
-            .ForMember(dest => dest.DayOffTypeId, opt => opt.MapFrom(src => Convert.ToInt32(src.DayOffType)))
+            .ForMember(dest => dest.DayOffTypeId, opt => opt.MapFrom(src => Convert.ToInt32(src.DayOffType.Id)))
             .ForMember(dest => dest.DayOffPayTypeId, opt => opt.MapFrom(src => Convert.ToInt32(src.DayOffPayType)));
 
         CreateMap<DayOffRequest, DayOffRequestModel>()
-        .ForMember(dest => dest.DayOffType, opt => opt.MapFrom(src => (DayOffTypeModel)src.DayOffTypeId))
```
Comment (samsonkassa1) [inline Telonai/TelonaiBFF/Helpers/AutoMapperProfile.cs:388]: I don't agree to this. I don't know why you had to change existing code to fetch dayofftype.

File: Telonai/TelonaiBFF/Helpers/DataContext.cs [hunk 4]
```diff
@@ -170,7 +171,5 @@ public virtual async Task<int> SaveChangesAsync()
     public DbSet<AgentFieldValue> AgentFieldValue { get; set; }
     public DbSet<MobileAppVersion> MobileAppVersion { get; set; }
     public DbSet<DayOffRequest> DayOffRequest { get; set; }
-    public DbSet<DayOffType> DayoffType { get; set; }
-    public DbSet<DayOffPayType> DayOffPayType { get; set; }
```
Comment (samsonkassa1) [inline Telonai/TelonaiBFF/Helpers/DataContext.cs:174]: Why the need to delete DayOffPayType?

File: Telonai/TelonaiBFF/Helpers/DayOffTypeModel.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+namespace TelonaiWebApi.Helpers;
```
Comment (samsonkassa1) [inline Telonai/TelonaiBFF/Helpers/DayOffTypeModel.cs:1]: There is already DayOffTypeModel. Why is the need for this one?

**Review by samsonkassa1 — COMMENTED** [review]
something is not right. You probably misunderstood the requirement. You don't need 16 file changes to add a single endpoint that uses existing models and entities. You only need one file change:
1. Create one httpGet method to return DayOfTypeModel enum strings as a list.

File: Telonai/TelonaiBFF/Entities/IdValuePair.cs (hunk 2 above)
Comment (biruk-telonai) [reply Telonai/TelonaiBFF/Entities/IdValuePair.cs:29]: fixed

File: Telonai/TelonaiBFF/Helpers/AutoMapperProfile.cs (hunk 3 above)
Comment (biruk-telonai) [reply Telonai/TelonaiBFF/Helpers/AutoMapperProfile.cs:388]: fixed

File: Telonai/TelonaiBFF/Helpers/DataContext.cs (hunk 4 above)
Comment (biruk-telonai) [reply Telonai/TelonaiBFF/Helpers/DataContext.cs:174]: fixed

**Review by biruk-telonai — COMMENTED** [review]
fixed

File: Telonai/TelonaiBFF/Controllers/DayOffRequestController.cs (hunk 1 above)
Comment (biruk-telonai) [reply Telonai/TelonaiBFF/Controllers/DayOffRequestController.cs]: fixed

File: Telonai/TelonaiBFF/Helpers/DayOffTypeModel.cs (hunk 5 above)
Comment (biruk-telonai) [reply Telonai/TelonaiBFF/Helpers/DayOffTypeModel.cs]: I have updated the name to DayOffTypeModel to DayOffTypes

**Pull request merged**